--- a/src/libuvc/dev.c	2017-01-20 21:56:31.000000000 +0000
+++ b/src/libuvc/dev.c	2018-04-27 17:24:09.565912713 +0100
@@ -36,6 +36,7 @@
  * @brief Support for finding, inspecting and opening UVC devices
  */
 
+#include <sys/sysmacros.h>
 #include "libuvc.h"
 #include "libuvc_internal.h"
 
--- a/src/uvc-v4l2.cpp	2017-01-20 21:56:31.000000000 +0000
+++ b/src/uvc-v4l2.cpp	2018-04-27 17:24:48.238302109 +0100
@@ -3,6 +3,7 @@
 
 #ifdef RS_USE_V4L2_BACKEND
 
+#include <sys/sysmacros.h>
 #include "uvc.h"
 
 #include <cassert>
