--- a/include/cob_obstacle_distance/link_to_collision.hpp
+++ b/include/cob_obstacle_distance/link_to_collision.hpp
@@ -47,15 +47,15 @@
 class LinkToCollision
 {
     private:
-        typedef boost::shared_ptr<const urdf::Link> PtrConstLink_t;
-        typedef boost::shared_ptr<urdf::Link> PtrLink_t;
+        typedef std::shared_ptr<const urdf::Link> PtrConstLink_t;
+        typedef std::shared_ptr<urdf::Link> PtrLink_t;
         typedef std::vector<boost::shared_ptr<urdf::Link> > VecPtrLink_t;
-        typedef boost::shared_ptr<urdf::Collision> PtrCollision_t;
-        typedef boost::shared_ptr<urdf::Geometry> PtrGeometry_t;
-        typedef boost::shared_ptr<urdf::Mesh> PtrMesh_t;
-        typedef boost::shared_ptr<urdf::Box> PtrBox_t;
-        typedef boost::shared_ptr<urdf::Sphere> PtrSphere_t;
-        typedef boost::shared_ptr<urdf::Cylinder> PtrCylinder_t;
+        typedef std::shared_ptr<urdf::Collision> PtrCollision_t;
+        typedef std::shared_ptr<urdf::Geometry> PtrGeometry_t;
+        typedef std::shared_ptr<urdf::Mesh> PtrMesh_t;
+        typedef std::shared_ptr<urdf::Box> PtrBox_t;
+        typedef std::shared_ptr<urdf::Sphere> PtrSphere_t;
+        typedef std::shared_ptr<urdf::Cylinder> PtrCylinder_t;
         typedef std::unordered_map<std::string, std::vector<std::string> >::iterator MapIter_t;
 
         urdf::Model model_;
-- 
2.17.0

