diff -r -u a/lookandfeel/contents/logout/Logout.qml b/lookandfeel/contents/logout/Logout.qml
--- a/lookandfeel/contents/logout/Logout.qml	2018-09-05 14:01:54.801253403 +0100
+++ b/lookandfeel/contents/logout/Logout.qml	2018-09-05 14:02:39.049714821 +0100
@@ -47,8 +47,12 @@
     function sleepRequested() {
         root.suspendRequested(2);
     }
- 
-    property real timeout: 30
+
+    function hibernateRequested() {
+        root.suspendRequested(4);
+    }
+
+    property real timeout: 10
     property real remainingTime: root.timeout
     property var currentAction: {
         switch (sdtype) {
@@ -132,11 +136,20 @@
             spacing: units.largeSpacing * 2
             Layout.alignment: Qt.AlignHCenter
             LogoutButton {
+                id: hibernateButton
+                iconSource: "system-suspend-hibernate"
+                text: i18nd("plasma_lookandfeel_org.kde.lookandfeel", "Hibernate")
+                action: root.hibernateRequested
+                KeyNavigation.left: logoutButton
+                KeyNavigation.right: suspendButton
+                visible: spdMethods.HibernateState
+            }
+            LogoutButton {
                 id: suspendButton
                 iconSource: "system-suspend"
                 text: i18nd("plasma_lookandfeel_org.kde.lookandfeel", "Suspend")
                 action: root.sleepRequested
-                KeyNavigation.left: logoutButton
+                KeyNavigation.left: hibernateButton
                 KeyNavigation.right: rebootButton
                 visible: spdMethods.SuspendState
             }
